packages/core/lib/use-puck.ts

```diff
@@ -16,9 +16,13 @@ export const usePuck = () => {
     appState,
     config,
     dispatch,
-    getPermissions: (selectedItem?: ComponentData) => {
+    getPermissions: (
+      selectedItem?: ComponentData,
+      type?: keyof DefaultComponentProps
```

Comment:
The `usePuck` hook should probably take a generic type of `UserConfig` (see other examples) so that the `type` can be fully typed against the user's actual components.

Reply:
Resolving and tracking in new issue: #579